### ISSUE TITLE
read `settings` config for rules like `remark-lint-strong-marker` 

### ISSUE THREAD

**Issue by JounQin** [remarkjs/remark-lint#263]
### Subject of the feature

As title

### Problem

```js
var report = require('vfile-reporter')
var remark = require('remark')
var emphasisMarker = require('remark-lint-emphasis-marker')
var strongMarker = require('remark-lint-strong-marker')

remark()
  .use({
    settings: { emphasis: '*', strong: '*' },
    // ^ `remark-stringify` settings.
  })
  .use(emphasisMarker)
  .use(strongMarker)
  // ^ two `remark-lint` rules.
  .process('_Hello_, __world__!', function(err, file) {
    console.error(report(err || file))
    console.log(String(file))
  })
```

There is no warning at all.

### Expected behavior

`emphasisMarker` and `strongMarker` can read `remark-stringify` settings automatically to set the default style, so that the user don't have to config same thing twice.

### Alternatives

N/A

**Comment by wooorm** [comment]
That might work 🤔

Rules don’t have access to `settings` though. So that would mean some changes have to land in [`unified-lint-rule`](https://github.com/remarkjs/remark-lint/blob/main/packages/unified-lint-rule/index.js) first.

**Comment by wooorm** [comment]
I think it’s fine to do this. One part is:

https://github.com/remarkjs/remark-lint/blob/909d9cceea9007b8d4029c89c2209f428880914b/packages/unified-lint-rule/lib/index.js#L64

which should probably be:

```
}).call(this, tree, file, options)
```

then, `this.data('settings')` could be uses in the rules, based on https://github.com/syntax-tree/mdast-util-to-markdown#tomarkdowntree-options, to replace the default where needed.

@JounQin Is this something you could help with?

**Comment by JounQin** [comment]
@wooorm What kind help do you mean? Raising a PR as you suggested or something else?

**Comment by wooorm** [comment]
Yeah, if this is something you could and would want to work on?
I could perhaps also work on it?

**Comment by JounQin** [comment]
I'd like to help, but I'm a bit busy these days, and I'm not so familiar with the code base.

I'll let you know whether I have time to work on before tomorrow, is this OK?

**Comment by wooorm** [comment]
Of course, no rush!
Can also be in a week or so.
Or if you’re not interested, that’s also okay!

**Comment by wooorm** [comment]
I have thought about this for a while now and I don‘t think it is a good idea. Closing this!

First, in many cases, users pass serialization options without `settings`, but directly to `remark-stringify`. We cannot catch that:

```js
unified()
  .use(remarkParse)
  .use(remarkLintEmphasisMarker)
  .use(remarkLintStrongMarker)
  .use(remarkStringify, {emphasis: '*', strong: '*'})
  // …
```

Direct options are also preferred if both exist:

```
  .use({settings: {emphasis: '_', strong: '_'}})
  .use(remarkStringify, {emphasis: '*', strong: '*'})
```

Second, I think it becomes too complex to reason what the source of things is.
There are also users who want the *rule* to be the source,
for how it is configured or for what it finds in a document to create the settings,
to then affect `remark-stringify`.

Third, lint rules are about the input document, settings (in this case) about the output document.
They do not have to be the same — they could have different preferences.

Finally, `settings` is an exception. Normally plugins are given options. Those options are not shared with other plugins. But that cannot happen with `remark-parse` and `remark-stringify` when used through `remark`. *That’s* the exception for why `settings` exist and when to use it.
